lib/util/http/github.ts

```diff
@@ -172,6 +174,76 @@ function constructAcceptString(input?: any): string {
   return acceptStrings.join(', ');
 }
 
+const MAX_GRAPHQL_PAGE_SIZE = 100;
+
+function getGraphqlPageSize(
+  fieldName: string,
+  defaultPageSize = MAX_GRAPHQL_PAGE_SIZE
+): number {
+  const cache = getCache();
+  cache.platform ??= {};
+  cache.platform.github ??= {};
+  cache.platform.github.graphqlPageCache ??= {};
+  const cachedRecord = cache.platform.github.graphqlPageCache[fieldName];
+
+  if (cachedRecord) {
+    logger.debug(
+      { fieldName, ...cachedRecord },
+      'GraphQL page size: found cached value'
+    );
+
+    const oldPageSize = cachedRecord.pageSize;
+
+    const now = DateTime.local();
+    const then = DateTime.fromISO(cachedRecord.pageLastResizedAt);
+    const expiry = then.plus({ hours: 24 });
+    if (now > expiry) {
+      const newPageSize = Math.min(oldPageSize * 2, MAX_GRAPHQL_PAGE_SIZE);
+      if (newPageSize < MAX_GRAPHQL_PAGE_SIZE) {
+        const timestamp = now.toISO();
+
+        logger.debug(
+          { fieldName, oldPageSize, newPageSize, timestamp },
+          'GraphQL page size: expanding'
+        );
+
+        cache.platform.github.graphqlPageCache[fieldName] = {
+          pageLastResizedAt: timestamp,
+          pageSize: newPageSize,
+        };
+
+        return newPageSize;
+      }
+    }
+
+    return oldPageSize;
+  }
+
+  return defaultPageSize;
+}
+
+function setGraphqlPageSize(fieldName: string, newPageSize: number): void {
+  if (newPageSize < MAX_GRAPHQL_PAGE_SIZE) {
```

Comment:
The idea is "don't bother caching if the page size is 100" ?

Reply:
Yes, though actually seems better to be outside